Fragments keeps reloading in a loop

### What version of Hls.js are you using?

1.5.16

### What browser (including version) are you using?

Chrome 129.0.6668.101 (Official Build) (64-bit)

### What OS (including version) are you using?

Windows 11

### Test stream

https://codepen.io/ShoobyD/details/OJKmPGp

### Configuration

```JavaScript
{}
```


### Additional player setup steps

_No response_

### Checklist

- [X] The issue observed is not already reported by searching on Github under https://github.com/video-dev/hls.js/issues
- [X] The issue occurs in the stable client (latest release) on https://hlsjs.video-dev.org/demo and not just on my page
- [X] The issue occurs in the latest client (main branch) on https://hlsjs-dev.video-dev.org/demo and not just on my page
- [X] The stream has correct Access-Control-Allow-Origin headers (CORS)
- [X] There are no network errors such as 404s in the browser console when trying to play the stream

### Steps to reproduce

1. Just attach this stream using the library:
'https://testing1eus2video-mv.azureedge.net/pictures/41/265/41265023/featured/low/108055_vl2c0jr8w75a/vid.m3u8'
No need to actually play it.

### Expected behaviour

The video fragments should load once.

### What actually happened?

The next fragment is being loaded over and over in a loop:
![hls js fragment loading issue](https://github.com/user-attachments/assets/a3cc1a97-11b4-46a6-8c5c-1b8d342270e5)

Even if you navigate to the middle of the video, it will load its current position fragment (let's say 'vid_1_3.ts'), and then load the next one ('vid_1_4.ts') again and again:
![hls js fragment loading issue 2](https://github.com/user-attachments/assets/31d31c19-665e-49de-8424-bfb131fbc493)





### Console output

```shell
main.js:1535 [getEditorValue] could not parse editor value SyntaxError: Expected double-quoted property name in JSON at position 19 (line 3 column 1)
    at JSON.parse (<anonymous>)
    at getEditorValue (main.js:1533:20)
    at loadSelectedStream (main.js:337:5)
    at HTMLInputElement.<anonymous> (main.js:142:5)
    at HTMLInputElement.dispatch (jquery.min.js:3:6444)
    at r.handle (jquery.min.js:3:3219)
getEditorValue @ main.js:1535
loadSelectedStream @ main.js:337
(anonymous) @ main.js:142
dispatch @ jquery.min.js:3
r.handle @ jquery.min.js:3
main.js:350 Using Hls.js config: {debug: true, enableWorker: true, lowLatencyMode: true, backBufferLength: 90}
logger.ts:74 [log] > Debug logs enabled for "Hls instance" in hls.js version 1.5.16
hls.ts:442 [log] > stopLoad
hls.ts:410 [log] > loadSource:https://testing1eus2video-mv.azureedge.net/pictures/41/265/41265023/featured/low/108055_vl2c0jr8w75a/vid.m3u8
stream-controller.ts:576 [log] > [stream-controller]: Trigger BUFFER_RESET
hls.ts:380 [log] > attachMedia
buffer-controller.ts:198 [log] > [buffer-controller] created media source: MediaSource
buffer-controller.ts:994 [log] > [buffer-controller] Media source opened
level-controller.ts:337 [log] > [level-controller]: manifest loaded, 2 level(s) found, first bitrate: 149286500
abr-controller.ts:58 [log] > setting initial bwe to 5000000
buffer-controller.ts:186 [log] > [buffer-controller] 1 bufferCodec event(s) expected
abr-controller.ts:701 [log] > [abr] picked start tier {"codecSet":"avc1,mp4a","videoRanges":["SDR"],"preferHDR":false,"minFramerate":0,"minBitrate":99558800}
abr-controller.ts:855 [info] > [abr] switch candidate:1->0 adjustedbw(99558800)-bitrate=0 ttfb:0.1 avgDuration:0.0 maxFetchDuration:4.0 fetchDuration:0.1 firstSelection:true codecSet:avc1,mp4a videoRange:SDR hls.loadLevel:-1
hls.ts:431 [log] > startLoad(-1)
level-controller.ts:432 [log] > [level-controller]: Switching to level 0 (720p SDR avc1,mp4a @99558800) from level -1
level-controller.ts:600 [log] > [level-controller]: Loading level index 0 with https://testing1eus2video-mv.azureedge.net/pictures/41/265/41265023/featured/low/108055_vl2c0jr8w75a/vid_1.m3u8
base-stream-controller.ts:1801 [log] > [stream-controller]: STOPPED->IDLE
base-stream-controller.ts:1801 [log] > [subtitle-stream-controller]: STOPPED->IDLE
stream-controller.ts:639 [log] > [stream-controller]: Level 0 loaded [0,5][part-5--1], cc [0, 0] duration:60.633334
buffer-controller.ts:861 [log] > [buffer-controller] Updating Media Source duration to 60.633
base-stream-controller.ts:758 [log] > [stream-controller]: Loading fragment 0 cc: 0 of [0-5] level: 0, target: 0
base-stream-controller.ts:1801 [log] > [stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:88 [log] > injecting Web Worker for "main"
transmuxer-interface.ts:230 [log] > [transmuxer-interface, main]: Starting new transmux session for sn: 0 p: -1 level: 0 id: 1
        discontinuity: true
        trackSwitch: true
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 0
        initSegmentChange: true
base-stream-controller.ts:389 [log] > [stream-controller]: Loaded fragment 0 of level 0
9be023df-1177-4cc4-b1c9-460e1422e1c1:555 [log] > Debug logs enabled for "main" in hls.js version 1.5.16
transmuxer-interface.ts:394 [log] > [mp4-remuxer]: ISGenerated flag reset
transmuxer-interface.ts:394 [log] > [mp4-remuxer]: initPTS & initDTS reset
transmuxer-interface.ts:394 [log] > [mp4-remuxer]: reset next timestamp
transmuxer-interface.ts:394 [log] > manifest codec:mp4a.40.2, ADTS type:2, samplingIndex:3
transmuxer-interface.ts:394 [log] > parsed codec:mp4a.40.5, rate:48000, channels:2
base-stream-controller.ts:1801 [log] > [stream-controller]: FRAG_LOADING->PARSING
stream-controller.ts:1303 [log] > [stream-controller]: Init audio buffer, container:audio/mp4, codecs[selected/level/parsed]=[mp4a.40.2/mp4a.40.2/mp4a.40.5]
stream-controller.ts:1314 [log] > [stream-controller]: Init video buffer, container:video/mp4, codecs[level/parsed]=[avc1.640033/avc1.640033]
buffer-controller.ts:393 [log] > [buffer-controller] 0 bufferCodec event(s) expected audio,video
buffer-controller.ts:943 [log] > [buffer-controller] creating sourceBuffer(audio/mp4;codecs=mp4a.40.2)
buffer-controller.ts:943 [log] > [buffer-controller] creating sourceBuffer(video/mp4;codecs=avc1.640033)
audio-stream-controller.ts:134 [log] > [audio-stream-controller]: InitPTS for cc: 0 found from main: 130080
transmuxer-interface.ts:394 [log] > [transmuxer.ts]: Flushed fragment 0 of level 0
base-stream-controller.ts:1801 [log] > [stream-controller]: PARSING->PARSED
base-stream-controller.ts:570 [log] > [stream-controller]: Buffered main sn: 0 of level 0 (frag:[0.000-10.888] > buffer:[0.021-10.837])
base-stream-controller.ts:1801 [log] > [stream-controller]: PARSED->IDLE
base-stream-controller.ts:758 [log] > [stream-controller]: Loading fragment 1 cc: 0 of [0-5] level: 0, target: 10.837
base-stream-controller.ts:1801 [log] > [stream-controller]: IDLE->FRAG_LOADING
base-stream-controller.ts:389 [log] > [stream-controller]: Loaded fragment 1 of level 0
base-stream-controller.ts:1801 [log] > [stream-controller]: FRAG_LOADING->PARSING
buffer-operation-queue.ts:64 [warn] > [buffer-operation-queue]: Exception executing "video" SourceBuffer operation: QuotaExceededError: Failed to execute 'appendBuffer' on 'SourceBuffer': The SourceBuffer is full, and cannot free space to append additional buffers.
executeNext @ buffer-operation-queue.ts:64
append @ buffer-operation-queue.ts:29
onBufferAppending @ buffer-controller.ts:548
emit @ index.js:203
emit @ hls.ts:310
trigger @ hls.ts:318
bufferFragmentData @ base-stream-controller.ts:943
_handleTransmuxComplete @ stream-controller.ts:1187
handleTransmuxComplete @ transmuxer-interface.ts:422
onWorkerMessage @ transmuxer-interface.ts:382
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:91
Show 1 more frame
Show less
base-stream-controller.ts:1043 [warn] > [stream-controller]: Reduce max buffer length to 300s
reduceMaxBufferLength @ base-stream-controller.ts:1043
reduceLengthAndFlushBuffer @ base-stream-controller.ts:1601
onError @ stream-controller.ts:923
emit @ index.js:203
emit @ hls.ts:310
trigger @ hls.ts:318
onError @ buffer-controller.ts:545
executeNext @ buffer-operation-queue.ts:67
append @ buffer-operation-queue.ts:29
onBufferAppending @ buffer-controller.ts:548
emit @ index.js:203
emit @ hls.ts:310
trigger @ hls.ts:318
bufferFragmentData @ base-stream-controller.ts:943
_handleTransmuxComplete @ stream-controller.ts:1187
handleTransmuxComplete @ transmuxer-interface.ts:422
onWorkerMessage @ transmuxer-interface.ts:382
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:91
Show 2 more frames
Show less
base-stream-controller.ts:1659 [log] > [stream-controller]: Reset loading state
base-stream-controller.ts:1801 [log] > [stream-controller]: PARSING->IDLE
main.js:745 Error event: {type: 'mediaError', parent: 'main', details: 'bufferFullError', sourceBufferName: 'video', frag: Fragment, …}
(anonymous) @ main.js:745
emit @ index.js:203
emit @ hls.ts:310
trigger @ hls.ts:318
onError @ buffer-controller.ts:545
executeNext @ buffer-operation-queue.ts:67
append @ buffer-operation-queue.ts:29
onBufferAppending @ buffer-controller.ts:548
emit @ index.js:203
emit @ hls.ts:310
trigger @ hls.ts:318
bufferFragmentData @ base-stream-controller.ts:943
_handleTransmuxComplete @ stream-controller.ts:1187
handleTransmuxComplete @ transmuxer-interface.ts:422
onWorkerMessage @ transmuxer-interface.ts:382
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:91
Show 2 more frames
Show less
transmuxer-interface.ts:394 [log] > [transmuxer.ts]: Flushed fragment 1 of level 0
base-stream-controller.ts:758 [log] > [stream-controller]: Loading fragment 1 cc: 0 of [0-5] level: 0, target: 10.837
base-stream-controller.ts:1801 [log] > [stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:230 [log] > [transmuxer-interface, main]: Starting new transmux session for sn: 1 p: -1 level: 0 id: 1
        discontinuity: false
        trackSwitch: false
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 10.837333333333333
        initSegmentChange: false
base-stream-controller.ts:389 [log] > [stream-controller]: Loaded fragment 1 of level 0
transmuxer-interface.ts:394 [log] > [mp4-remuxer]: reset next timestamp
base-stream-controller.ts:1801 [log] > [stream-controller]: FRAG_LOADING->PARSING
buffer-operation-queue.ts:64 [warn] > [buffer-operation-queue]: Exception executing "video" SourceBuffer operation: QuotaExceededError: Failed to execute 'appendBuffer' on 'SourceBuffer': The SourceBuffer is full, and cannot free space to append additional buffers.
executeNext @ buffer-operation-queue.ts:64
append @ buffer-operation-queue.ts:29
onBufferAppending @ buffer-controller.ts:548
emit @ index.js:203
emit @ hls.ts:310
trigger @ hls.ts:318
bufferFragmentData @ base-stream-controller.ts:943
_handleTransmuxComplete @ stream-controller.ts:1187
handleTransmuxComplete @ transmuxer-interface.ts:422
onWorkerMessage @ transmuxer-interface.ts:382
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:91
Show 1 more frame
Show less
base-stream-controller.ts:1043 [warn] > [stream-controller]: Reduce max buffer length to 150s
reduceMaxBufferLength @ base-stream-controller.ts:1043
reduceLengthAndFlushBuffer @ base-stream-controller.ts:1601
onError @ stream-controller.ts:923
emit @ index.js:203
emit @ hls.ts:310
trigger @ hls.ts:318
onError @ buffer-controller.ts:545
executeNext @ buffer-operation-queue.ts:67
append @ buffer-operation-queue.ts:29
onBufferAppending @ buffer-controller.ts:548
emit @ index.js:203
emit @ hls.ts:310
trigger @ hls.ts:318
bufferFragmentData @ base-stream-controller.ts:943
_handleTransmuxComplete @ stream-controller.ts:1187
handleTransmuxComplete @ transmuxer-interface.ts:422
onWorkerMessage @ transmuxer-interface.ts:382
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:91
Show 2 more frames
Show less
base-stream-controller.ts:1659 [log] > [stream-controller]: Reset loading state
base-stream-controller.ts:1801 [log] > [stream-controller]: PARSING->IDLE
main.js:745 Error event: {type: 'mediaError', parent: 'main', details: 'bufferFullError', sourceBufferName: 'video', frag: Fragment, …}
(anonymous) @ main.js:745
emit @ index.js:203
emit @ hls.ts:310
trigger @ hls.ts:318
onError @ buffer-controller.ts:545
executeNext @ buffer-operation-queue.ts:67
append @ buffer-operation-queue.ts:29
onBufferAppending @ buffer-controller.ts:548
emit @ index.js:203
emit @ hls.ts:310
trigger @ hls.ts:318
bufferFragmentData @ base-stream-controller.ts:943
_handleTransmuxComplete @ stream-controller.ts:1187
handleTransmuxComplete @ transmuxer-interface.ts:422
onWorkerMessage @ transmuxer-interface.ts:382
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:91
Show 2 more frames
Show less
transmuxer-interface.ts:394 [log] > [transmuxer.ts]: Flushed fragment 1 of level 0
base-stream-controller.ts:758 [log] > [stream-controller]: Loading fragment 1 cc: 0 of [0-5] level: 0, target: 10.837
base-stream-controller.ts:1801 [log] > [stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:230 [log] > [transmuxer-interface, main]: Starting new transmux session for sn: 1 p: -1 level: 0 id: 1
        discontinuity: false
        trackSwitch: false
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 10.837333333333333
        initSegmentChange: false
base-stream-controller.ts:389 [log] > [stream-controller]: Loaded fragment 1 of level 0
transmuxer-interface.ts:394 [log] > [mp4-remuxer]: reset next timestamp
base-stream-controller.ts:1801 [log] > [stream-controller]: FRAG_LOADING->PARSING
buffer-operation-queue.ts:64 [warn] > [buffer-operation-queue]: Exception executing "video" SourceBuffer operation: QuotaExceededError: Failed to execute 'appendBuffer' on 'SourceBuffer': The SourceBuffer is full, and cannot free space to append additional buffers.
executeNext @ buffer-operation-queue.ts:64
append @ buffer-operation-queue.ts:29
onBufferAppending @ buffer-controller.ts:548
emit @ index.js:203
emit @ hls.ts:310
trigger @ hls.ts:318
bufferFragmentData @ base-stream-controller.ts:943
_handleTransmuxComplete @ stream-controller.ts:1187
handleTransmuxComplete @ transmuxer-interface.ts:422
onWorkerMessage @ transmuxer-interface.ts:382
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:91
Show 1 more frame
Show less
base-stream-controller.ts:1043 [warn] > [stream-controller]: Reduce max buffer length to 75s
reduceMaxBufferLength @ base-stream-controller.ts:1043
reduceLengthAndFlushBuffer @ base-stream-controller.ts:1601
onError @ stream-controller.ts:923
emit @ index.js:203
emit @ hls.ts:310
trigger @ hls.ts:318
onError @ buffer-controller.ts:545
executeNext @ buffer-operation-queue.ts:67
append @ buffer-operation-queue.ts:29
onBufferAppending @ buffer-controller.ts:548
emit @ index.js:203
emit @ hls.ts:310
trigger @ hls.ts:318
bufferFragmentData @ base-stream-controller.ts:943
_handleTransmuxComplete @ stream-controller.ts:1187
handleTransmuxComplete @ transmuxer-interface.ts:422
onWorkerMessage @ transmuxer-interface.ts:382
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:91
Show 2 more frames
Show less
base-stream-controller.ts:1659 [log] > [stream-controller]: Reset loading state
base-stream-controller.ts:1801 [log] > [stream-controller]: PARSING->IDLE
main.js:745 Error event: {type: 'mediaError', parent: 'main', details: 'bufferFullError', sourceBufferName: 'video', frag: Fragment, …}
(anonymous) @ main.js:745
emit @ index.js:203
emit @ hls.ts:310
trigger @ hls.ts:318
onError @ buffer-controller.ts:545
executeNext @ buffer-operation-queue.ts:67
append @ buffer-operation-queue.ts:29
onBufferAppending @ buffer-controller.ts:548
emit @ index.js:203
emit @ hls.ts:310
trigger @ hls.ts:318
bufferFragmentData @ base-stream-controller.ts:943
_handleTransmuxComplete @ stream-controller.ts:1187
handleTransmuxComplete @ transmuxer-interface.ts:422
onWorkerMessage @ transmuxer-interface.ts:382
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:91
Show 2 more frames
Show less
transmuxer-interface.ts:394 [log] > [transmuxer.ts]: Flushed fragment 1 of level 0
base-stream-controller.ts:758 [log] > [stream-controller]: Loading fragment 1 cc: 0 of [0-5] level: 0, target: 10.837
base-stream-controller.ts:1801 [log] > [stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:230 [log] > [transmuxer-interface, main]: Starting new transmux session for sn: 1 p: -1 level: 0 id: 1
        discontinuity: false
        trackSwitch: false
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 10.837333333333333
        initSegmentChange: false
base-stream-controller.ts:389 [log] > [stream-controller]: Loaded fragment 1 of level 0
transmuxer-interface.ts:394 [log] > [mp4-remuxer]: reset next timestamp
base-stream-controller.ts:1801 [log] > [stream-controller]: FRAG_LOADING->PARSING
buffer-operation-queue.ts:64 [warn] > [buffer-operation-queue]: Exception executing "video" SourceBuffer operation: QuotaExceededError: Failed to execute 'appendBuffer' on 'SourceBuffer': The SourceBuffer is full, and cannot free space to append additional buffers.
executeNext @ buffer-operation-queue.ts:64
append @ buffer-operation-queue.ts:29
onBufferAppending @ buffer-controller.ts:548
emit @ index.js:203
emit @ hls.ts:310
trigger @ hls.ts:318
bufferFragmentData @ base-stream-controller.ts:943
_handleTransmuxComplete @ stream-controller.ts:1187
handleTransmuxComplete @ transmuxer-interface.ts:422
onWorkerMessage @ transmuxer-interface.ts:382
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:91
Show 1 more frame
Show less
base-stream-controller.ts:1043 [warn] > [stream-controller]: Reduce max buffer length to 37.5s
reduceMaxBufferLength @ base-stream-controller.ts:1043
reduceLengthAndFlushBuffer @ base-stream-controller.ts:1601
onError @ stream-controller.ts:923
emit @ index.js:203
emit @ hls.ts:310
trigger @ hls.ts:318
onError @ buffer-controller.ts:545
executeNext @ buffer-operation-queue.ts:67
append @ buffer-operation-queue.ts:29
onBufferAppending @ buffer-controller.ts:548
emit @ index.js:203
emit @ hls.ts:310
trigger @ hls.ts:318
bufferFragmentData @ base-stream-controller.ts:943
_handleTransmuxComplete @ stream-controller.ts:1187
handleTransmuxComplete @ transmuxer-interface.ts:422
onWorkerMessage @ transmuxer-interface.ts:382
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:91
Show 2 more frames
Show less
base-stream-controller.ts:1659 [log] > [stream-controller]: Reset loading state
base-stream-controller.ts:1801 [log] > [stream-controller]: PARSING->IDLE
main.js:745 Error event: {type: 'mediaError', parent: 'main', details: 'bufferFullError', sourceBufferName: 'video', frag: Fragment, …}chunkMeta: {level: 0, sn: 1, part: -1, id: 1, size: 95149996, …}details: "bufferFullError"err: QuotaExceededError: Failed to execute 'appendBuffer' on 'SourceBuffer': The SourceBuffer is full, and cannot free space to append additional buffers.
    at BufferController.appendExecutor (https://hlsjs.video-dev.org/dist/hls.js:19168:10)
    at Object.execute (https://hlsjs.video-dev.org/dist/hls.js:18673:18)
    at BufferOperationQueue.executeNext (https://hlsjs.video-dev.org/dist/hls.js:18249:21)
    at BufferOperationQueue.append (https://hlsjs.video-dev.org/dist/hls.js:18220:14)
    at BufferController.onBufferAppending (https://hlsjs.video-dev.org/dist/hls.js:18741:22)
    at EventEmitter.emit (https://hlsjs.video-dev.org/dist/hls.js:15633:36)
    at Hls.emit (https://hlsjs.video-dev.org/dist/hls.js:28593:28)
    at Hls.trigger (https://hlsjs.video-dev.org/dist/hls.js:28597:21)
    at StreamController.bufferFragmentData (https://hlsjs.video-dev.org/dist/hls.js:10169:16)
    at StreamController._handleTransmuxComplete (https://hlsjs.video-dev.org/dist/hls.js:28199:16)error: QuotaExceededError: Failed to execute 'appendBuffer' on 'SourceBuffer': The SourceBuffer is full, and cannot free space to append additional buffers.
    at BufferController.appendExecutor (https://hlsjs.video-dev.org/dist/hls.js:19168:10)
    at Object.execute (https://hlsjs.video-dev.org/dist/hls.js:18673:18)
    at BufferOperationQueue.executeNext (https://hlsjs.video-dev.org/dist/hls.js:18249:21)
    at BufferOperationQueue.append (https://hlsjs.video-dev.org/dist/hls.js:18220:14)
    at BufferController.onBufferAppending (https://hlsjs.video-dev.org/dist/hls.js:18741:22)
    at EventEmitter.emit (https://hlsjs.video-dev.org/dist/hls.js:15633:36)
    at Hls.emit (https://hlsjs.video-dev.org/dist/hls.js:28593:28)
    at Hls.trigger (https://hlsjs.video-dev.org/dist/hls.js:28597:21)
    at StreamController.bufferFragmentData (https://hlsjs.video-dev.org/dist/hls.js:10169:16)
    at StreamController._handleTransmuxComplete (https://hlsjs.video-dev.org/dist/hls.js:28199:16)errorAction: {action: 0, flags: 0}fatal: falsefrag: Fragment {_byteRange: null, _url: 'https://testing1eus2video-mv.azureedge.net/picture…65023/featured/low/108055_vl2c0jr8w75a/vid_1_1.ts', baseurl: 'https://testing1eus2video-mv.azureedge.net/picture…65023/featured/low/108055_vl2c0jr8w75a/vid_1.m3u8', relurl: 'vid_1_1.ts', elementaryStreams: {…}, …}parent: "main"part: nullsourceBufferName: "video"type: "mediaError"[[Prototype]]: Object
(anonymous) @ main.js:745
emit @ index.js:203
emit @ hls.ts:310
trigger @ hls.ts:318
onError @ buffer-controller.ts:545
executeNext @ buffer-operation-queue.ts:67
append @ buffer-operation-queue.ts:29
onBufferAppending @ buffer-controller.ts:548
emit @ index.js:203
emit @ hls.ts:310
trigger @ hls.ts:318
bufferFragmentData @ base-stream-controller.ts:943
_handleTransmuxComplete @ stream-controller.ts:1187
handleTransmuxComplete @ transmuxer-interface.ts:422
onWorkerMessage @ transmuxer-interface.ts:382
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:91
Show 2 more frames
Show less
transmuxer-interface.ts:394 [log] > [transmuxer.ts]: Flushed fragment 1 of level 0
base-stream-controller.ts:758 [log] > [stream-controller]: Loading fragment 1 cc: 0 of [0-5] level: 0, target: 10.837
base-stream-controller.ts:1801 [log] > [stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:230 [log] > [transmuxer-interface, main]: Starting new transmux session for sn: 1 p: -1 level: 0 id: 1
        discontinuity: false
        trackSwitch: false
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 10.837333333333333
        initSegmentChange: false
base-stream-controller.ts:389 [log] > [stream-controller]: Loaded fragment 1 of level 0
transmuxer-interface.ts:394 [log] > [mp4-remuxer]: reset next timestamp
base-stream-controller.ts:1801 [log] > [stream-controller]: FRAG_LOADING->PARSING
buffer-operation-queue.ts:64 [warn] > [buffer-operation-queue]: Exception executing "video" SourceBuffer operation: QuotaExceededError: Failed to execute 'appendBuffer' on 'SourceBuffer': The SourceBuffer is full, and cannot free space to append additional buffers.
executeNext @ buffer-operation-queue.ts:64
append @ buffer-operation-queue.ts:29
onBufferAppending @ buffer-controller.ts:548
emit @ index.js:203
emit @ hls.ts:310
trigger @ hls.ts:318
bufferFragmentData @ base-stream-controller.ts:943
_handleTransmuxComplete @ stream-controller.ts:1187
handleTransmuxComplete @ transmuxer-interface.ts:422
onWorkerMessage @ transmuxer-interface.ts:382
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:91
Show 1 more frame
Show less
base-stream-controller.ts:1043 [warn] > [stream-controller]: Reduce max buffer length to 18.75s
reduceMaxBufferLength @ base-stream-controller.ts:1043
reduceLengthAndFlushBuffer @ base-stream-controller.ts:1601
onError @ stream-controller.ts:923
emit @ index.js:203
emit @ hls.ts:310
trigger @ hls.ts:318
onError @ buffer-controller.ts:545
executeNext @ buffer-operation-queue.ts:67
append @ buffer-operation-queue.ts:29
onBufferAppending @ buffer-controller.ts:548
emit @ index.js:203
emit @ hls.ts:310
trigger @ hls.ts:318
bufferFragmentData @ base-stream-controller.ts:943
_handleTransmuxComplete @ stream-controller.ts:1187
handleTransmuxComplete @ transmuxer-interface.ts:422
onWorkerMessage @ transmuxer-interface.ts:382
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:91
Show 2 more frames
Show less
base-stream-controller.ts:1659 [log] > [stream-controller]: Reset loading state
base-stream-controller.ts:1801 [log] > [stream-controller]: PARSING->IDLE
main.js:745 Error event: {type: 'mediaError', parent: 'main', details: 'bufferFullError', sourceBufferName: 'video', frag: Fragment, …}
(anonymous) @ main.js:745
emit @ index.js:203
emit @ hls.ts:310
trigger @ hls.ts:318
onError @ buffer-controller.ts:545
executeNext @ buffer-operation-queue.ts:67
append @ buffer-operation-queue.ts:29
onBufferAppending @ buffer-controller.ts:548
emit @ index.js:203
emit @ hls.ts:310
trigger @ hls.ts:318
bufferFragmentData @ base-stream-controller.ts:943
_handleTransmuxComplete @ stream-controller.ts:1187
handleTransmuxComplete @ transmuxer-interface.ts:422
onWorkerMessage @ transmuxer-interface.ts:382
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:91
Show 2 more frames
Show less
base-stream-controller.ts:758 [log] > [stream-controller]: Loading fragment 1 cc: 0 of [0-5] level: 0, target: 10.837
base-stream-controller.ts:1801 [log] > [stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:394 [log] > [transmuxer.ts]: Flushed fragment 1 of level 0
base-stream-controller.ts:1801 [log] > [stream-controller]: FRAG_LOADING->PARSING
base-stream-controller.ts:1801 [log] > [stream-controller]: PARSING->PARSED
base-stream-controller.ts:570 [log] > [stream-controller]: Buffered main sn: 1 of level 0 (frag:[10.837-20.032] > buffer:[0.021-10.837])
base-stream-controller.ts:1801 [log] > [stream-controller]: PARSED->IDLE
abr-controller.ts:607 [info] > [abr] rebuffering expected, optimal quality level -1
base-stream-controller.ts:758 [log] > [stream-controller]: Loading fragment 1 cc: 0 of [0-5] level: 0, target: 10.837
base-stream-controller.ts:1801 [log] > [stream-controller]: IDLE->FRAG_LOADING
transmuxer-interface.ts:230 [log] > [transmuxer-interface, main]: Starting new transmux session for sn: 1 p: -1 level: 0 id: 1
        discontinuity: false
        trackSwitch: false
        contiguous: false
        accurateTimeOffset: true
        timeOffset: 10.837333333333333
        initSegmentChange: false
base-stream-controller.ts:389 [log] > [stream-controller]: Loaded fragment 1 of level 0
transmuxer-interface.ts:394 [log] > [mp4-remuxer]: reset next timestamp
base-stream-controller.ts:1801 [log] > [stream-controller]: FRAG_LOADING->PARSING
buffer-operation-queue.ts:64 [warn] > [buffer-operation-queue]: Exception executing "video" SourceBuffer operation: QuotaExceededError: Failed to execute 'appendBuffer' on 'SourceBuffer': The SourceBuffer is full, and cannot free space to append additional buffers.
executeNext @ buffer-operation-queue.ts:64
append @ buffer-operation-queue.ts:29
onBufferAppending @ buffer-controller.ts:548
emit @ index.js:203
emit @ hls.ts:310
trigger @ hls.ts:318
bufferFragmentData @ base-stream-controller.ts:943
_handleTransmuxComplete @ stream-controller.ts:1187
handleTransmuxComplete @ transmuxer-interface.ts:422
onWorkerMessage @ transmuxer-interface.ts:382
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:91
Show 1 more frame
Show less
base-stream-controller.ts:1043 [warn] > [stream-controller]: Reduce max buffer length to 9.375s
reduceMaxBufferLength @ base-stream-controller.ts:1043
reduceLengthAndFlushBuffer @ base-stream-controller.ts:1601
onError @ stream-controller.ts:923
emit @ index.js:203
emit @ hls.ts:310
trigger @ hls.ts:318
onError @ buffer-controller.ts:545
executeNext @ buffer-operation-queue.ts:67
append @ buffer-operation-queue.ts:29
onBufferAppending @ buffer-controller.ts:548
emit @ index.js:203
emit @ hls.ts:310
trigger @ hls.ts:318
bufferFragmentData @ base-stream-controller.ts:943
_handleTransmuxComplete @ stream-controller.ts:1187
handleTransmuxComplete @ transmuxer-interface.ts:422
onWorkerMessage @ transmuxer-interface.ts:382
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:91
Show 2 more frames
Show less
base-stream-controller.ts:1659 [log] > [stream-controller]: Reset loading state
base-stream-controller.ts:1801 [log] > [stream-controller]: PARSING->IDLE
main.js:745 Error event: {type: 'mediaError', parent: 'main', details: 'bufferFullError', sourceBufferName: 'video', frag: Fragment, …}
(anonymous) @ main.js:745
emit @ index.js:203
emit @ hls.ts:310
trigger @ hls.ts:318
onError @ buffer-controller.ts:545
executeNext @ buffer-operation-queue.ts:67
append @ buffer-operation-queue.ts:29
onBufferAppending @ buffer-controller.ts:548
emit @ index.js:203
emit @ hls.ts:310
trigger @ hls.ts:318
bufferFragmentData @ base-stream-controller.ts:943
_handleTransmuxComplete @ stream-controller.ts:1187
handleTransmuxComplete @ transmuxer-interface.ts:422
onWorkerMessage @ transmuxer-interface.ts:382
TransmuxerInterface.onwmsg @ transmuxer-interface.ts:91
Show 2 more frames
Show less
transmuxer-interface.ts:394 [log] > [transmuxer.ts]: Flushed fragment 1 of level 0
abr-controller.ts:607 [info] > [abr] rebuffering expected, optimal quality level -1
abr-controller.ts:607 [info] > [abr] rebuffering expected, optimal quality level -1
abr-controller.ts:607 [info] > [abr] rebuffering expected, optimal quality level -1
abr-controller.ts:607 [info] > [abr] rebuffering expected, optimal quality level -1
abr-controller.ts:607 [info] > [abr] rebuffering expected, optimal quality level -1
abr-controller.ts:607 [info] > [abr] rebuffering expected, optimal quality level -1
abr-controller.ts:607 [info] > [abr] rebuffering expected, optimal quality level -1
abr-controller.ts:607 [info] > [abr] rebuffering expected, optimal quality level -1
abr-controller.ts:607 [info] > [abr] rebuffering expected, optimal quality level -1
abr-controller.ts:607 [info] > [abr] rebuffering expected, optimal quality level -1
abr-controller.ts:607 [info] > [abr] rebuffering expected, optimal quality level -1
abr-controller.ts:607 [info] > [abr] rebuffering expected, optimal quality level -1
abr-controller.ts:607 [info] > [abr] rebuffering expected, optimal quality level -1
abr-controller.ts:607 [info] > [abr] rebuffering expected, optimal quality level -1
abr-controller.ts:607 [info] > [abr] rebuffering expected, optimal quality level -1
abr-controller.ts:607 [info] > [abr] rebuffering expected, optimal quality level -1
abr-controller.ts:607 [info] > [abr] rebuffering expected, optimal quality level -1
.....
```


### Chrome media internals output

_No response_

--- a/api-extractor/report/hls.js.api.md
+++ b/api-extractor/report/hls.js.api.md
@@ -1987,6 +1987,7 @@ export class FragmentTracker implements ComponentAPI {
     // (undocumented)
     fragBuffered(frag: MediaFragment, force?: true): void;
     getAppendedFrag(position: number, levelType: PlaylistLevelType): MediaFragment | Part | null;
+    getBackBufferEvictionEnd(beforePosition: number, levelType: PlaylistLevelType, bytesNeeded: number): number;
     getBufferedFrag(position: number, levelType: PlaylistLevelType): MediaFragment | null;
     // (undocumented)
     getFragAtPos(position: number, levelType: PlaylistLevelType, buffered?: boolean): MediaFragment | null;

--- a/src/controller/buffer-controller.ts
+++ b/src/controller/buffer-controller.ts
@@ -118,6 +118,12 @@ export default class BufferController extends Logger implements ComponentAPI {
     audiovideo: 0,
   };
   private appendError?: ErrorData;
+  // Tracks whether a QuotaExceededError back-buffer eviction is in progress
+  // for a given SourceBuffer type. While true, subsequent QuotaExceededErrors
+  // on the same type piggyback on the pending eviction instead of triggering
+  // their own or emitting BUFFER_FULL_ERROR.
+  private _quotaEvictionPending: Partial<Record<SourceBufferName, boolean>> =
+    {};
   // Record of required or created buffers by type. SourceBuffer is stored in Track.buffer once created.
   private tracks: SourceBufferTrackSet = {};
   // Array of SourceBuffer type and SourceBuffer (or null). One entry per TrackSet in this.tracks.
@@ -853,6 +859,7 @@ transfer tracks: ${stringify(transferredTracks, (key, value) => (key === 'initSe
     }
 
     const fragStart = (part || frag).start;
+    let quotaEvictionAttempted = false;
     const operation: BufferOperation = {
       label: `append-${type}`,
       execute: () => {
@@ -900,6 +907,46 @@ transfer tracks: ${stringify(transferredTracks, (key, value) => (key === 'initSe
       },
       onError: (error: Error) => {
         this.clearBufferAppendTimeoutId(this.tracks[type]);
+
+        const isQuotaError =
+          (error as DOMException).code === DOMException.QUOTA_EXCEEDED_ERR ||
+          error.name == 'QuotaExceededError' ||
+          `quota` in error;
+
+        if (isQuotaError) {
+          // An eviction is already in flight for this type — piggyback on it
+          // by queuing a retry after the pending remove, no new eviction needed.
+          if (this._quotaEvictionPending[type]) {
+            this.log(
+              `QuotaExceededError on "${type}" sn: ${sn} — eviction already pending, queuing retry`,
+            );
+            this.insertNext([operation], type);
+            return;
+          }
+
+          // First QuotaExceededError for this type: evict minimum back buffer
+          if (!quotaEvictionAttempted) {
+            const evictEnd = this.getBackBufferEvictionTarget(
+              type,
+              data.byteLength,
+              frag.type,
+            );
+            if (evictEnd > 0) {
+              quotaEvictionAttempted = true;
+              this._quotaEvictionPending[type] = true;
+              this.log(
+                `QuotaExceededError on "${type}" append sn: ${sn} — evicting back buffer to ${evictEnd.toFixed(3)}s and retrying`,
+              );
+              const removeOp = this.getQuotaEvictionFlushOp(type, 0, evictEnd);
+              this.insertNext([removeOp, operation], type);
+              return;
+            }
+            this.warn(
+              `QuotaExceededError on "${type}" sn: ${sn} — no back buffer available to evict`,
+            );
+          }
+        }
+
         // in case any error occured while appending, put back segment in segments table
         const event: ErrorData = {
           type: ErrorTypes.MEDIA_ERROR,
@@ -914,13 +961,9 @@ transfer tracks: ${stringify(transferredTracks, (key, value) => (key === 'initSe
           fatal: false,
         };
         const mediaError = this.media?.error;
-        if (
-          (error as DOMException).code === DOMException.QUOTA_EXCEEDED_ERR ||
-          error.name == 'QuotaExceededError' ||
-          `quota` in error
-        ) {
+        if (isQuotaError) {
           // QuotaExceededError: http://www.w3.org/TR/html5/infrastructure.html#quotaexceedederror
-          // let's stop appending any segments, and report BUFFER_FULL_ERROR error
+          // Eviction was already attempted or not possible — report BUFFER_FULL_ERROR
           event.details = ErrorDetails.BUFFER_FULL_ERROR;
         } else if (
           (error as DOMException).code === DOMException.INVALID_STATE_ERR &&
@@ -969,6 +1012,33 @@ transfer tracks: ${stringify(transferredTracks, (key, value) => (key === 'initSe
       } ${frag.level} cc: ${cc} offset: ${offset} bytes: ${data.byteLength}`,
     );
     this.append(operation, type, this.isPending(this.tracks[type]));
+  }
+
+  // Like getFlushOp but clears _quotaEvictionPending on complete/error
+  private getQuotaEvictionFlushOp(
+    type: SourceBufferName,
+    start: number,
+    end: number,
+  ): BufferOperation {
+    this.log(`queuing quota-eviction "${type}" remove ${start}-${end}`);
+    return {
+      label: 'remove',
+      execute: () => {
+        this.removeExecutor(type, start, end);
+      },
+      onStart: () => {},
+      onComplete: () => {
+        this._quotaEvictionPending[type] = false;
+        this.hls.trigger(Events.BUFFER_FLUSHED, { type });
+      },
+      onError: (error: Error) => {
+        this._quotaEvictionPending[type] = false;
+        this.warn(
+          `Failed to remove ${start}-${end} from "${type}" SourceBuffer (quota eviction)`,
+          error,
+        );
+      },
+    };
   }
 
   private getFlushOp(
@@ -1196,6 +1266,26 @@ transfer tracks: ${stringify(transferredTracks, (key, value) => (key === 'initSe
         this.resetAppendErrors();
       }
     }
+  }
+
+  // Returns the end position to evict back buffer to on QuotaExceededError,
+  // or 0 if there is nothing to evict. Delegates to the fragment tracker which
+  // walks buffered fragments oldest-first using actual byte sizes to find the
+  // minimum eviction needed to fit the new segment.
+  private getBackBufferEvictionTarget(
+    type: SourceBufferName,
+    segmentBytes: number,
+    playlistType: PlaylistLevelType,
+  ): number {
+    const { media } = this;
+    if (!media) {
+      return 0;
+    }
+    return this.fragmentTracker.getBackBufferEvictionEnd(
+      media.currentTime,
+      playlistType,
+      segmentBytes,
+    );
   }
 
   private resetAppendErrors() {
@@ -1937,6 +2027,12 @@ transfer tracks: ${stringify(transferredTracks, (key, value) => (key === 'initSe
   ) {
     if (this.operationQueue) {
       this.operationQueue.append(operation, type, pending);
+    }
+  }
+
+  private insertNext(operations: BufferOperation[], type: SourceBufferName) {
+    if (this.operationQueue) {
+      this.operationQueue.insertNext(operations, type);
     }
   }
 

--- a/src/controller/buffer-operation-queue.ts
+++ b/src/controller/buffer-operation-queue.ts
@@ -79,6 +79,14 @@ export default class BufferOperationQueue {
     );
   }
 
+  public insertNext(operations: BufferOperation[], type: SourceBufferName) {
+    if (this.queues === null) {
+      return;
+    }
+    // Insert after the current (index 0) operation so they execute next
+    this.queues[type].splice(1, 0, ...operations);
+  }
+
   public unblockAudio(op: BufferOperation) {
     if (this.queues === null) {
       return;

--- a/src/controller/fragment-tracker.ts
+++ b/src/controller/fragment-tracker.ts
@@ -483,13 +483,10 @@ export class FragmentTracker implements ComponentAPI {
   }
 
   /**
-   * Returns the eviction end position needed to free at least `bytesNeeded`
-   * from the back buffer, or 0 if not enough data is available.
-   * Walks buffered fragments oldest-first, accumulating actual byte sizes.
-   *
-   * Uses stats.loaded (actual bytes received) as the primary byte measure
-   * since it is always set after fragment completion, unlike stats.total
-   * which depends on Content-Length and may be 0 with chunked transfer.
+   * Returns the end position needed to free at least `bytesNeeded` from the
+   * back buffer, or 0 if not enough data is available. Walks buffered
+   * fragments in key order, accumulating byte sizes using stats.loaded,
+   * byteLength, or a bitrate estimate as fallback.
    */
   public getBackBufferEvictionEnd(
     beforePosition: number,
@@ -499,8 +496,8 @@ export class FragmentTracker implements ComponentAPI {
     const { fragments } = this;
 
     // Collect back buffer fragments with known byte sizes
-    let count = 0;
-    const candidates: FragmentEntity[] = [];
+    let bytesFreed = 0;
+    let evictEnd = 0;
     const keys = Object.keys(fragments);
     for (let i = 0; i < keys.length; i++) {
       const entity = fragments[keys[i]];
@@ -509,32 +506,16 @@ export class FragmentTracker implements ComponentAPI {
       }
       const frag = entity.body;
       // Use stats.loaded (always set after load) with byteLength as fallback
-      const bytes = (frag.hasStats && frag.stats.loaded) || frag.byteLength;
+      const bytes =
+        (frag.hasStats && frag.stats.loaded) ||
+        frag.byteLength ||
+        (frag.bitrate && frag.bitrate * 8 * frag.duration);
       if (frag.end <= beforePosition && bytes) {
-        candidates[count++] = entity;
-      }
-    }
-
-    if (count === 0) {
-      return 0;
-    }
-
-    // Sort by end time (earliest first). Using frag.end rather than sn because
-    // the tracker may contain fragments from different HLS variants where
-    // media sequence numbers are not guaranteed to align across levels.
-    candidates.length = count;
-    candidates.sort((a, b) => a.body.end - b.body.end);
-
-    // Walk oldest-first, accumulating bytes until we have enough
-    let bytesFreed = 0;
-    let evictEnd = 0;
-    for (let i = 0; i < count; i++) {
-      const frag = candidates[i].body;
-      const bytes = (frag.hasStats && frag.stats.loaded) || frag.byteLength;
-      bytesFreed += bytes!;
-      evictEnd = frag.end;
-      if (bytesFreed >= bytesNeeded) {
-        return evictEnd;
+        bytesFreed += bytes;
+        evictEnd = Math.max(evictEnd, frag.end);
+        if (bytesFreed >= bytesNeeded) {
+          return evictEnd;
+        }
       }
     }
 

--- a/src/controller/fragment-tracker.ts
+++ b/src/controller/fragment-tracker.ts
@@ -482,6 +482,68 @@ export class FragmentTracker implements ComponentAPI {
     return !!this.activePartLists[type]?.length;
   }
 
+  /**
+   * Returns the eviction end position needed to free at least `bytesNeeded`
+   * from the back buffer, or 0 if not enough data is available.
+   * Walks buffered fragments oldest-first, accumulating actual byte sizes.
+   *
+   * Uses stats.loaded (actual bytes received) as the primary byte measure
+   * since it is always set after fragment completion, unlike stats.total
+   * which depends on Content-Length and may be 0 with chunked transfer.
+   */
+  public getBackBufferEvictionEnd(
+    beforePosition: number,
+    levelType: PlaylistLevelType,
+    bytesNeeded: number,
+  ): number {
+    const { fragments } = this;
+
+    // Collect back buffer fragments with known byte sizes
+    let count = 0;
+    const candidates: FragmentEntity[] = [];
+    const keys = Object.keys(fragments);
+    for (let i = 0; i < keys.length; i++) {
+      const entity = fragments[keys[i]];
+      if (!entity || !entity.buffered || entity.body.type !== levelType) {
+        continue;
+      }
+      const frag = entity.body;
+      // Use stats.loaded (always set after load) with byteLength as fallback
+      const bytes = (frag.hasStats && frag.stats.loaded) || frag.byteLength;
+      if (frag.end <= beforePosition && bytes) {
+        candidates[count++] = entity;
+      }
+    }
+
+    if (count === 0) {
+      return 0;
+    }
+
+    // Sort by sn (oldest first) — small array of references, no data copying.
+    // Candidates are unsorted because this.fragments is a hash map keyed by
+    // "type_level_sn", and Object.keys() returns insertion order which depends
+    // on when each fragment was loaded, not its position in the timeline.
+    // ABR level switches and seeks can cause fragments to load out of order.
+    candidates.length = count;
+    candidates.sort((a, b) => (a.body.sn as number) - (b.body.sn as number));
+
+    // Walk oldest-first, accumulating bytes until we have enough
+    let bytesFreed = 0;
+    let evictEnd = 0;
+    for (let i = 0; i < count; i++) {
+      const frag = candidates[i].body;
+      const bytes = (frag.hasStats && frag.stats.loaded) || frag.byteLength;
+      bytesFreed += bytes!;
+      evictEnd = frag.end;
+      if (bytesFreed >= bytesNeeded) {
+        return evictEnd;
+      }
+    }
+
+    // Not enough to fully cover bytesNeeded, return what we have
+    return evictEnd > 0 ? evictEnd : 0;
+  }
+
   public removeFragmentsInRange(
     start: number,
     end: number,

--- a/src/controller/fragment-tracker.ts
+++ b/src/controller/fragment-tracker.ts
@@ -519,13 +519,11 @@ export class FragmentTracker implements ComponentAPI {
       return 0;
     }
 
-    // Sort by sn (oldest first) — small array of references, no data copying.
-    // Candidates are unsorted because this.fragments is a hash map keyed by
-    // "type_level_sn", and Object.keys() returns insertion order which depends
-    // on when each fragment was loaded, not its position in the timeline.
-    // ABR level switches and seeks can cause fragments to load out of order.
+    // Sort by end time (earliest first). Using frag.end rather than sn because
+    // the tracker may contain fragments from different HLS variants where
+    // media sequence numbers are not guaranteed to align across levels.
     candidates.length = count;
-    candidates.sort((a, b) => (a.body.sn as number) - (b.body.sn as number));
+    candidates.sort((a, b) => a.body.end - b.body.end);
 
     // Walk oldest-first, accumulating bytes until we have enough
     let bytesFreed = 0;

